tools/analysis_tools/analyze_logs.py

```diff
@@ -17,18 +17,20 @@ def cal_train_time(log_dicts, args):
                 all_times.append(log_dict[epoch]['time'])
             else:
                 all_times.append(log_dict[epoch]['time'][1:])
-        all_times = np.array(all_times)
-        epoch_ave_time = all_times.mean(-1)
+        if not all_times:
+            raise KeyError(
+                'Please reduce the log interval in the config so that '
+                'interval is less than iterations of one epoch.')
+        epoch_ave_time = np.array(list(map(lambda x: np.mean(x), all_times)))
```

Comment:
This modification which uses iteration may be slower than the prior code?

Reply:
Original implementation will raise a `TypeError`.